### PR TITLE
Major Refactor

### PR DESCRIPTION
**Pull request by ChineduOguejiofor**
#### What does this PR achieve?
- Updates the auth endpoints
- Updates the nearest vendors endpoint
- Updates the user/vendor Profile endpoint

#### Any background context?
- For the auth endpoints, we were storing the information on two different collections which gave us several issues when retrieving the data. The two collections have been combined to one
- Several routes had similar idea.

#### Relevant Trello Tickets

- [Ticket 1](some.com)

### CODE REVIEW

File: functions/controllers/auth.js [hunk 1]
```diff
@@ -89,10 +89,10 @@ exports.addUser = async (req, res, next) => {
...
         infoData.createdAt = formatDate(infoData.createdAt);
 
-        const addressSnapshot = await db.doc(`addresses/${userAuth.uid}`).get();
-        if (addressSnapshot.exists) {
-            infoData.address = addressSnapshot.data();
-        }
+        // const addressSnapshot = await db.doc(`addresses/${userAuth.uid}`).get();
+        // if (addressSnapshot.exists) {
+        //     infoData.address = addressSnapshot.data();
+        // }
```
Comment (edbella) [inline functions/controllers/auth.js]: Remove unused codes

File: functions/config/cleanit.js [hunk 2]
```diff
@@ -9,6 +9,24 @@ const adminConfig = {
...
+    storageBucket: process.env.STORAGE_BUCKET,
+    messagingSenderId: process.env.MESSAGING_SENDER_ID,
+    appId: process.env.APP_ID,
+    measurementId: process.env.MEASUREMENT_ID,
+    private_key: process.env.PRIVATE_KEY,
+};
+module.exports = {
+    adminConfig,
+    firebaseConfig,
+>>>>>>> Updated Vendors list API, Updated auth endpoint
```
Comment (Way2nnadi) [inline functions/config/cleanit.js]: merge conflict errors on this page

File: functions/controllers/auth.js [hunk 3]
```diff
@@ -13,53 +13,62 @@ const sendEmailhelper = async email => {
...
+<<<<<<< HEAD
     let userRef;
     if (userType === 'user') {
         userRef = db.doc(`users/${userAuth.uid}`);
     } else if (userType === 'vendor') {
         userRef = db.doc(`vendors/${userAuth.uid}`);    
     } else {
+=======
+    if (userType !== 'user' && userType !== 'vendor') {
+>>>>>>> Completed Refactor all routes are functional
```
Comment (Way2nnadi) [inline functions/controllers/auth.js]: merge conflict errors

File: functions/controllers/auth.js [hunk 4]
```diff
@@ -13,53 +13,62 @@ const sendEmailhelper = async email => {
...
+        let name;
+        if (!snapShot.exists) {
+            const {
+                email, displayName, photoURL, emailVerified, phoneNumber,
+            } = userAuth;
+            if (!displayName) {
+                name = fullName;
+            } else if (!fullName) {
+                name = displayName;
+            }
```
Comment (Way2nnadi) [inline functions/controllers/auth.js]: one liner `const name = ! displayName ? fullName : displayName;`

File: functions/controllers/profile.js [hunk 5]
```diff
@@ -24,18 +26,45 @@ const getUserInfo = async (userType, userID) => {
...
+            }
+        });
+    } catch (error) {
+        return 'unable to communitcate with database';
+    }
+};
+
+const addToGeoCollection = async (addresses, vendorID) => {
+    await deleteVendors(vendorID);
+    for (let i = 0; i < addresses.length; i += 1) {
```
Comment (Way2nnadi) [inline functions/controllers/profile.js]: use `for...of` loop

File: functions/controllers/vendor.js [hunk 6]
```diff
@@ -1,28 +1,70 @@
...
 
-exports.vendorAddress = async (req, res, next) => {
-    const { address, vendorLat, vendorLng } = req.body;
-    const createdAt = new Date();
-    const vendorID = req.authId;
+        for (let i = 0; i < closestVendors.length; i += 1) {
+            const currentVendor = closestVendors[i];
+            const vendorInfo = await getVendorInfo(currentVendor.vendorID);
+            currentVendor.vendorInfo = vendorInfo;
+            closestVendors[i] = currentVendor;
```
Comment (Way2nnadi) [inline functions/controllers/vendor.js]: do not do this, create a new variable and a new array to store the updated data.

File: functions/controllers/vendor.js [hunk 7]
```diff
@@ -1,28 +1,70 @@
...
+                vendorID,
+            };
+            closestVendors.push(queryDetails);
+        });
 
-exports.vendorAddress = async (req, res, next) => {
-    const { address, vendorLat, vendorLng } = req.body;
-    const createdAt = new Date();
-    const vendorID = req.authId;
+        for (let i = 0; i < closestVendors.length; i += 1) {
```
Comment (Way2nnadi) [inline functions/controllers/vendor.js]: use a `Array.map` here
```
const updatedVendors = closestVendors.map(...)
```

File: functions/config/cleanit.js (hunk 2 above)
Comment (ChineduOguejiofor) [reply functions/config/cleanit.js]: What an oversight I didn't test after rebasing 
Sorry about this

File: functions/controllers/auth.js (hunk 3 above)
Comment (ChineduOguejiofor) [reply functions/controllers/auth.js]: Same here

File: functions/controllers/auth.js (hunk 4 above)
Comment (ChineduOguejiofor) [reply functions/controllers/auth.js]: Updated

File: functions/controllers/profile.js (hunk 5 above)
Comment (ChineduOguejiofor) [reply functions/controllers/profile.js]: Updated

File: functions/controllers/vendor.js (hunk 7 above)
Comment (ChineduOguejiofor) [reply functions/controllers/vendor.js]: I am not sure how to achieve this

File: functions/controllers/vendor.js (hunk 6 above)
Comment (ChineduOguejiofor) [reply functions/controllers/vendor.js]: having some issues with this

File: functions/controllers/vendor.js (hunk 7 above)
Comment (ChineduOguejiofor) [reply functions/controllers/vendor.js]: map returns an empty obj even when I have added async

Comment (ChineduOguejiofor) [reply functions/controllers/vendor.js]: So I used the previous comment which was to use for of

File: functions/controllers/vendor.js (hunk 6 above)
Comment (ChineduOguejiofor) [reply functions/controllers/vendor.js]: I have addressed this in the for of loop

**Pull request merged**